spec/requests/demo_mode/sessions_spec.rb

```diff
@@ -62,6 +64,7 @@
               persona_name: 'the_everyperson',
               variant: 'alternate bruce',
             },
+            options: { example_custom_option: 'this is an example' },
```

Comment:
Are you thinking something like; if the persona is created successfully, return the options as part of the response as well?

Reply:
Since we are only enabling options for the post request, is there a good way to simulate the params within system spec? Or if we are testing it through request spec, is there a way to access the generated user? I don't have a good way to verify the changes I made within the `around_persona_generation` block.

Reply:
For example: I can update the user's last name within request spec, but I don't have access to the user context within the spec itself.
```
before do
          DemoMode.configure do  
            options_allow_list { %i(example_custom_option) }
    
            around_persona_generation do |generator, options|
              user = generator.call
        
              if options.present? && options[:last_name].present?
                user.update!(last_name: options[:last_name])
              end
      
              user
            end
          end
        end
```

Reply:
If this persona generates a `DummyUser`, there shouldn't be any `DummyUser` records at the beginning of the spec, so I think you could:
1. Have the persona app accept and persist an option for an attribute like `name`
2. Make the POST with that option param
3. Execute the job associated with the session ID from the response body to generate the persona
4. Query `DummyUser` for the only record, which should have the provided `name` attribute value

So we would have to drop down a level and query `DummyUser` instead of observing it entirely through network requests, but I think that's still a very valuable test. I don't think we're losing much anyway since directly testing anything after persona generation through the dummy app feels kinda like we're just testing the dummy app, not the demo mode / personas framework.

Does that sound reasonable?